libres/lib/enkf/ensemble_config.cpp

```diff
@@ -946,8 +946,8 @@ ensemble_config_add_container(ensemble_config_type *ensemble_config,
     char *local_key = (char *)key;
     bool random_key = false;
     if (key == NULL) {
-        local_key = (char *)util_calloc(11, sizeof *local_key);
-        sprintf(local_key, "%ld", random() % 10000000);
+        local_key = (char *)util_calloc(33, sizeof *local_key);
```

Comment:
So there is a nifty `std:format` for this in C++20, but we likely need to compile for some old software so the best is likely to use `stringstream`:
```cpp
std:string local_key = ("ensemble_config_random" << (random() % 10000000)).str();
...
enkf_config_node_type *config_node =
            enkf_config_node_new_container(local_key.c_str());
```